### ISSUE TITLE
[test][data grid] aria-labels missing from mui-x

### ISSUE THREAD

**Issue by annawatson-wk** [mui/mui-x#15074]
### Steps to reproduce

Steps:
1. Open https://mui.com/x/react-data-grid/#mit-license-free-forever
1. Inspect the column header component.
2. Note there are no aria-labels on mui-x components.

### Current behavior

MUI X components don't have aria-labels anymore. 

<img width="736" alt="SCR-20241024-beuc" src="https://github.com/user-attachments/assets/8e98807c-ee62-446a-bb5f-d13397d1eaca">

https://mui.com/x/react-data-grid/#mit-license-free-forever

### Expected behavior

MUI X components should have aria-labels.

<img width="762" alt="SCR-20241024-beky" src="https://github.com/user-attachments/assets/f18507d5-ad78-42b4-80cb-d81aca7222bc">

https://v6.mui.com/x/react-data-grid/#mit-version-free-forever

### Context

We had a couple of unit tests fail accessibility-wise for `columnheader` when upgrading to the latest version of mui-x from 7.13.0. Looking at it, it appears as though all the aria-labels from Data Grid are gone. Browsing the mui website, it looks like Date Picker is also effected. Is this intentional? 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 13.6.2
  Binaries:
    Node: 20.8.1 - ~/.asdf/installs/nodejs/20.8.1/bin/node
    npm: 10.1.0 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 129.0.6668.71
```
</details>

**Search keywords**: aria-label aria label data grid pro date picker
**Order ID**: 82849

**Comment by annawatson-wk** [comment]
I'm not sure if this work is potentially related or not: https://github.com/mui/mui-x/pull/14197

**Comment by oliviertassinari** [comment]
The a11y tree look correct:

<img width="285" alt="SCR-20241024-cucw" src="https://github.com/user-attachments/assets/b69b0844-d725-42ce-9dc6-edcbfc72cd01">

so it looks like this change is right, it's a simplification. It's about how to migrate the tests. I guess you are looking for using:

```jsx
screen.getByRole('columnheader', { name: /first name/i })
```

this should be invariant (work in both MUI X v6 and v7)

**Comment by annawatson-wk** [comment]
Brilliant, that will work. We just wanted to make sure it was intentional.

**Comment by KenanYusuf** [comment]
Glad to hear it @annawatson-wk - I'll close the issue but feel free to re-open if you have any follow up questions.

**Comment by oliviertassinari** [comment]
I think this was closed a bit prematurely.

> We just wanted to make sure it was intentional.

@annawatson-wk Yes, it fixes a bug, see #14482

> We had a couple of unit tests fail accessibility-wise for columnheader when upgrading to the latest version of mui-x from 7.13.0.

Could you share how you changed your tests? This woud hopefully help the next developers migrating. Thanks

**Comment by annawatson-wk** [comment]
> accessibility-wise
This was probably a poor choice of words. We had a consumer unit testing aria-labels.

**Comment by oliviertassinari** [comment]
@annawatson-wk Alright, if you have more details about the pickers aria-label change, or test change, we can continue to discuss this, otherwise, it looks like we can close.
